service/collector/aws.go

```diff
@@ -24,9 +26,16 @@ func (c Collector) getARNs() ([]string, error) {
 	arnsMap := make(map[string]bool)
 	for _, awsConfig := range awsConfigs.Items {
 		arn, err := credential.GetARN(c.k8sClient, &awsConfig)
-		if err != nil {
-			return nil, microerror.Mask(err)
+		// Collect as many ARNs as possible in order to provide most metrics.
+		// Ignore old cluster which do not have credential.
+		if err == nil {
```

Comment:
Ignoring errors is not a good thing. In case there is a specific error we should handle it and ignore the expected one only. All other errors should be dealt with though.

Reply:
Added specific error when credential is not there. PTAL